### ISSUE TITLE
DuplicateAction duplicate deep does not refresh elements lft or right values after previous structure is updated

### ISSUE THREAD

**Issue by Anubarak** [craftcms/cms#9889]
### Description

TL/DR
When you duplicate entries in a deep nested structure it will fetch the elements once at the beginning for each structure but won't update the elements->lft and right values after changes. So all elements have wrong lft and right values thus children cannot be found.
<hr>

![image](https://user-images.githubusercontent.com/6122122/135080209-bf6b14c9-dce6-42fa-a820-2bdddd08eb49.png)
The resulting duplicates are like
![image](https://user-images.githubusercontent.com/6122122/135080236-9815a2f4-6703-4ce2-8c32-044ee68a4726.png)
So it misses certain children.

This is because after a structure is moved the lft and right values of the elements will be updated and no children are found.
I logged everything in the Duplicate Action

> Run _duplicateElements
> Title: R1, lft: 121, right: 164
> Title: R2, lft: 177, right: 184
> Title: R3, lft: 197, right: 198
> Check for Children R1 left 121 right 164
> found 3 Children
> // ......
> // process these children.....
> Check for Children R2 left 177 right 184 // <-- old lft and right values here... the real values differ because other entries are already saved

When I change your code from
```PHP
if ($this->deep) {
    $children = $element->getChildren()->anyStatus()->all();
    $this->_duplicateElements($children, $successCount, $failCount, $duplicatedElementIds, $duplicate);
}
```

to
```PHP
if ($this->deep) {
    // refetch the new entry with correct lft and right values in order get the refreshed children
    $original = $element::find()->structureId($element->structureId)->id($element->id)->one();
    $children = $original->getChildren()->anyStatus()->all();
    $this->_duplicateElements($children, $successCount, $failCount, $duplicatedElementIds, $duplicate);
}
```

It works in most cases - but still not all unfortunately.. I had no time to figure out why it works for some structures but not for all.
```PHP 
$original = $element::find()->structureId($element->structureId)->id($element->id)->one();
$original->lft !== $element->lft
```

### Steps to reproduce

1. Create 3 level nested structure
2. Duplicate the elements -> see not all are duplicated because deep level children are not fetched due to the changed `lft` and `right` values

### Additional info

- Craft version: 3.7.12

**Comment by brandonkelly** [comment]
> It works in most cases - but still not all unfortunately.. I had no time to figure out why it works for some structures but not for all.

At first glance your suggested fix seems pretty close to what it should be. Could just be that you’re missing status & site params:

```php
$original = $element::find()
    ->structureId($element->structureId)
    ->id($element->id)
    ->siteId($element->siteId)
    ->anyStatus()
    ->one();
```

**Comment by brandonkelly** [comment]
Should be fixed for the next release.

To get the fix early, change your `craftcms/cms` requirement in composer.json to `"dev-develop as 3.7.14"` and run `composer update`.

**Comment by brandonkelly** [comment]
Craft 3.7.15 is out now with this fix.
